nixos/modules/virtualisation/amazon-image.nix

```diff
@@ -33,23 +33,28 @@ in
 
     boot.growPartition = cfg.hvm;
 
-    fileSystems."/" = {
+    fileSystems."/" = mkIf (!cfg.zfs.enable) {
       device = "/dev/disk/by-label/nixos";
       fsType = "ext4";
       autoResize = true;
     };
 
-    fileSystems."/boot" = mkIf cfg.efi {
+    fileSystems."/boot" = if cfg.efi then {
       device = "/dev/disk/by-label/ESP";
       fsType = "vfat";
-    };
+    } else if cfg.zfs.enable then {
+      device = "boot/boot";
+      fsType = "zfs";
+    } else {};
+
+    boot.zfs.devNodes = mkIf cfg.zfs.enable "/dev/";
 
     boot.extraModulePackages = [
       config.boot.kernelPackages.ena
     ];
     boot.initrd.kernelModules = [ "xen-blkfront" "xen-netfront" ];
     boot.initrd.availableKernelModules = [ "ixgbevf" "ena" "nvme" ];
-    boot.kernelParams = mkIf cfg.hvm [ "console=ttyS0" "random.trust_cpu=on" ];
+    boot.kernelParams = mkIf cfg.hvm [ "console=ttyS0,115200n8" "random.trust_cpu=on" ];
```

Comment:
This is not a ZFS change. Seems to have slipped in within another commit.

The docs states [Default is "9600n8"](https://github.com/torvalds/linux/blob/62fb9874f5da54fdb243003b386128037319b219/Documentation/admin-guide/kernel-parameters.txt#L646-L657) but uuuuh... 

Many of those entries, especially amba-pl* (spec-compliant ARM serial) actually define different defaults.

```
 ~/tmp/linux/linux $ ag 'int\s*baud\s*=\s*\d+' drivers/tty/serial/ | cat | sort
drivers/tty/serial/21285.c:423: int baud = 9600;
drivers/tty/serial/8250/8250_port.c:3211:       int baud = 9600;
drivers/tty/serial/amba-pl010.c:653:    int baud = 38400;
drivers/tty/serial/amba-pl011.c:2271:   int baud = 38400;
drivers/tty/serial/apbuart.c:479:       int baud = 38400;
drivers/tty/serial/ar933x_uart.c:583:   int baud = 115200;
drivers/tty/serial/arc_uart.c:494:      int baud = 115200;
drivers/tty/serial/atmel_serial.c:2524: int baud = 115200;
drivers/tty/serial/bcm63xx_uart.c:735:  int baud = 9600;
drivers/tty/serial/bfin_sport_uart.c:616:       int baud = 57600;
drivers/tty/serial/bfin_uart.c:1107:    int baud = 57600;
drivers/tty/serial/clps711x.c:389:      int baud = 38400, bits = 8, parity = 'n', flow = 'n';
drivers/tty/serial/cpm_uart/cpm_uart_core.c:1297:       int baud = 38400;
drivers/tty/serial/digicolor-usart.c:422:       int baud = 115200, bits = 8, parity = 'n', flow = 'n';
drivers/tty/serial/dz.c:867:    int baud = 9600;
drivers/tty/serial/efm32-uart.c:592:    int baud = 115200;
drivers/tty/serial/etraxfs-uart.c:93:   int baud = 115200;
drivers/tty/serial/fsl_lpuart.c:1779:   int baud = 115200;
drivers/tty/serial/imx.c:1918:  int baud = 9600;
drivers/tty/serial/ip22zilog.c:1034:    int baud = 9600, bits = 8;
drivers/tty/serial/lantiq.c:620:        int baud = 115200;
drivers/tty/serial/lpc32xx_hs.c:167:    int baud = 115200;
drivers/tty/serial/m32r_sio.c:983:      int baud = 9600;
drivers/tty/serial/max3100.c:432:       int baud = 0;
drivers/tty/serial/meson_uart.c:529:    int baud = 115200;
drivers/tty/serial/mps2-uart.c:424:     int baud = 9600;
drivers/tty/serial/msm_serial.c:1639:   int baud = 115200;
drivers/tty/serial/mvebu-uart.c:495:    int baud = 9600;
drivers/tty/serial/mxs-auart.c:1442:    int baud = 9600;
drivers/tty/serial/netx-serial.c:596:   int baud = 9600;
drivers/tty/serial/omap-serial.c:1303:  int baud = 115200;
drivers/tty/serial/pic32_uart.c:729:    int baud = 115200;
drivers/tty/serial/pmac_zilog.c:1998:   int baud = 38400;
drivers/tty/serial/pnx8xxx_uart.c:728:  int baud = 38400;
drivers/tty/serial/pxa.c:733:   int baud = 9600;
drivers/tty/serial/sa1100.c:784:        int baud = 9600;
drivers/tty/serial/samsung.c:2114:      int baud = 9600;
drivers/tty/serial/sb1250-duart.c:878:  int baud = 115200;
drivers/tty/serial/sccnxp.c:828:        int baud = 9600, bits = 8, parity = 'n', flow = 'n';
drivers/tty/serial/serial_core.c:2323:  int baud = 9600;
drivers/tty/serial/serial_ks8695.c:617: int baud = 115200;
drivers/tty/serial/serial_txx9.c:937:   int baud = 9600;
drivers/tty/serial/sh-sci.c:2827:       int baud = 115200;
drivers/tty/serial/sirfsoc_uart.c:1087: unsigned int baud = 115200;
drivers/tty/serial/sprd_serial.c:565:   int baud = 115200;
drivers/tty/serial/st-asc.c:837:        int baud = 9600;
drivers/tty/serial/stm32-usart.c:961:   int baud = 9600;
drivers/tty/serial/uartlite.c:478:      int baud = 9600;
drivers/tty/serial/vr41xx_siu.c:788:    int baud = 9600;
drivers/tty/serial/vt8500_serial.c:532: int baud = 9600;
drivers/tty/serial/xilinx_uartps.c:1261:        int baud = 9600;
drivers/tty/serial/zs.c:1203:   int baud = 9600;
```

If the change is desired, can this be done in a non-ZFS PR, or at the very least a distinct commit with the *reason* the serial default was changed?

Reply:
Yep, can do, thanks!

Reply:
Resolved in #134981, thanks.